homeassistant/components/vizio/media_player.py

```diff
@@ -136,6 +135,12 @@ def __init__(self, host, token, name, volume_step, device_type):
         self._device = Vizio(DEVICE_ID, host, DEFAULT_NAME, token, device_type)
         self._max_volume = float(self._device.get_max_volume())
 
+        esn = self._device.get_esn()
+        if esn:
+            self._unique_id = f"{device_type}-{esn}"
+        else:
+            self._unique_id = f"{device_type}-{slugify(host)}"
```

Comment:
IP addresses (`host`) should not be used to create a unique ID. You could just make it `None`.

Reply:
Wouldn't that potentially result in non-unique unique IDs? This is really a fallback in case the API call doesn't work because I couldn't test it on a Vizio TV, just a soundbar. My thought was that the IP address was the most secondarily unique thing about the device beyond ESN and MAC, and I couldn't think of a good way to get the MAC

Reply:
Setting it to `None` effectively means that it doesn't have a unique ID, so it wouldn't be added to the entity registry. But if you can't get a unique identifier, then what can you do? Feel free to ask someone more knowledgeable about HA than me if this is the correct approach, but that's what I was told for the Android TV integration. 

https://github.com/home-assistant/home-assistant/pull/22996#discussion_r275135940

Reply:
I see, thanks for the reference and the feedback. Updated accordingly.

Reply:
I do wonder how you can possibly use the entity registry for something like pihole... I added a PR for that one which uses the IP/port because I couldn't come up with any other unique identifiers.